### PR TITLE
fix: Use static Reflect methods in nodeGlobalThis proxy

### PR DESCRIPTION
**Pull request by kamilogorek**
```
  get(_target, prop, _receiver) {
    if (prop in nodeGlobals) {
      return nodeGlobals[prop];
    } else {
      return globalThis[prop];
    }
  },
```

For some reason, this block was getting confused, and when missing property (eg. `RegExp`) was asked for, `globalThis[prop]` pointed to the same proxied object, which then recursed infinitely.
One solution was to just use `target[prop]` instead, however for the sake of completeness, I changed all methods to use `Reflect` instead, as `ownKeys`, `defineProperty` and `getOwnPropertyDescriptor` already did that.

I cannot come up with an explicit test-case (didnt invest much time into it yet), but so far all current tests are passing and linked issue is fixed when trying repro locally.

Fixes https://github.com/denoland/deno/issues/17678

### CODE REVIEW

**Comment by kamilogorek** [comment]
One thing worth mentioning is that this issue does not occur when loading the module in-directly via `createRequire`, which may point in the direction of `npm:` specifiers doing some magic.

eg. this works just fine, even without this fix

```js
import { createRequire } from "https://deno.land/std@0.177.0/node/module.ts";
const require = createRequire(import.meta.url);
const prettier = require("./prettier");

prettier.format('', {filename: '1.js'});
```

File: ext/node/01_node.js [hunk 1]
```diff
@@ -27,62 +31,55 @@ function assert(cond) {
...
+  has(target, prop) {
+    return ReflectHas(nodeGlobals, prop) || ReflectHas(target, prop);
   },
-  ownKeys(_target) {
-    const globalThisKeys = ReflectOwnKeys(globalThis);
+  deleteProperty(target, prop) {
+    return (
+      ReflectDeleteProperty(nodeGlobals, prop) ||
+      ReflectDeleteProperty(target, prop)
+    );
```
Comment (dsherret) [inline ext/node/01_node.js:55]: This changes the logic. I've committed a test that passes on main, but fails on this PR.

File: cli/tests/testdata/npm/compare_globals/main.ts [hunk 2]
```diff
@@ -12,3 +12,16 @@ type _TestHasNodeJsGlobal = NodeJS.Architecture;
...
+// Node code has its own global and the Deno code has the same global,
+// but it's different. Basically if some Node code deletes
+// one of these globals then we don't want it to suddenly inherit
+// the Deno global.
+globals.withNodeGlobalThis((nodeGlobalThis: any) => {
+  (globalThis as any).setTimeout = 5;
+  console.log(setTimeout);
+  delete nodeGlobalThis["setTimeout"];
+  console.log(nodeGlobalThis["setTimeout"]); // should be undefined
+  console.log(globalThis["setTimeout"]); // should be undefined
```
Comment (dsherret) [inline cli/tests/testdata/npm/compare_globals/main.ts:26]: On main these both go to `undefined`. I'm not entirely sure if this one should, but that's the current behaviour

**Review by dsherret — COMMENTED** [review]
Thanks! This looks good other than the behaviour change in delete.

**Comment by kamilogorek** [comment]
Good catch, thanks! Updated

**Review by dsherret — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**